src/averageAges.js

```diff
@@ -15,11 +15,16 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const onlyMen = people.filter(men => men.sex === 'm');
+  const ages = onlyMen.map(year => year.died - year.born);
+
+  const agesCentury = onlyMen
+    .filter(men => Math.ceil(men.died / 100) === century)
+    .map(year => year.died - year.born);
```

Comment:
you did this logic (creating an array of ages 2 times)
let's process in next way: at the start if the function we'll define the correct array of people (depending on the century value), if it's present -> we need to find men with defined century, if no -> just all men

and then do average age calculation for one men array

Reply:
agree, stupid solution, will fix it! thx:)